backend/dal/Services/Concrete/ProjectReportService.cs

```diff
@@ -181,10 +181,13 @@ public ProjectReport Add(ProjectReport report)
         public ProjectReport Update(ProjectReport report)
         {
             this.User.ThrowIfNotAuthorized(Permissions.ReportsSpl);
+            var isSPLAdmin = this.User.HasPermission(Permissions.ReportsSplAdmin);
 
             var originalReport = this.Context.ProjectReports
                 .SingleOrDefault(p => p.Id == report.Id) ?? throw new KeyNotFoundException();
-
+            
+            if (!originalReport.IsFinal && report.IsFinal && !isSPLAdmin) throw new NotAuthorizedException ($"You do not have permissions to make a report final.");
```

Comment:
Probably could shrink this down to a single `if` statement.

Reply:
I removed the "extra" if statement that wasn't needed.

Reply:
This will allow a non admin to turn `IsFinal` off.  That was captured by the second `if` statement previously.  You should handle both scenarios in a single `if` statement.